apiary_v2.apib

```diff
@@ -2240,20 +2342,10 @@ endpoint provides:
             "manufacturing_files_zip": "https://factory.unmade.com/v2/jobs/e3dc59c3-d4ed-42bc-8d73-23dd0e5c4301/manufacturing_files_zip/",
             "ticket": "https://factory.unmade.com/v22/jobs/e3dc59c3-d4ed-42bc-8d73-23dd0e5c4301/ticket_pdf/",
             "design_source_id": "6489daff-b713-452c-b103-c724d307cb35",
```

Comment:
It turns out this is the [editor design](https://github.com/unmadeworks/embed-service/blob/731f19e9890b6f83ac6a384cf4a9c08ed083d6b7/embed_service/order/models.py#L755) rather than the one actually used to submit the manufacturing data.

This is fine for grouping purposes but might be confusing if you want to go back up to get data from the design endpoint for this job because you might be looking at a design with a different size (or the roster template).

Do you think we should also send down the order item design ID and show that one instead? We mention above that the `design_id` on the Order Item is the "source of all manufacturing data for this item", so it's that one we need to be showing here I think.

Reply:
It feels that the one currently is needed and make sense for grouping as you said. And if we want to offer the grouping to them we have it there. To keep or not to keep on the API (still not 100% sure), potentially maybe NB is relying on this one for grouping files themselves.
It might be that the current key is confusing? we could rename that key on v2 to indicate to the integrations that you should use those values for certain purposes.

If we want to allow from the manufacturing flow get more data about the design yes then we need that other design id. However, what's the flow that we're thinking where it will be needed? Or in other words what the design endpoint will give in terms of data that the job won't have: customisations added like logos, text, ... components chosen, materials? what has the design that the job doesn't?

I thought the manufacturing flow could be something like:
- I need an endpoint to retrieve a list of orders based on a period of time (clear and done)
- Per order, I want to retrieve: minimum information, the manufacturing files and other paperwork I need to manufacture: job ticket, order ticket, other paperwork, etc.

And I'm still on the fence about how and where should the API present back the "customisations" per job: placement choices, materials, components, print colours, etc .... 

Those things might be needed on the integration to build their own QC system (I believe that's what NB was after and using designs endpoint for) or grouping/sorting identifying etc ... and most of that data it might be already present on the Job.
 
I said "on the fence" just cause of the possibility of generating a big payload, still maybe not a big issue since we're planning on showing only X number of jobs for an order. 

If we feel that showing all that information it's going to be a problem, then we might need that extra endpoint, maybe something like `v2/orders/<order-id>/jobs-personalisation` or something like that.

Reply:
The existing field will definitely still be needed for the internal grouping. But I agree it's not necessarily needed on the public API.

I know what you mean about the flow. Linking from the job back up to the design is just a nice to have really, and not a requirement anyone has yet. We could actually just not show any Design IDs from the Factory side, and rely on putting relevant info in the job representation like you say. I don't think it needs its own endpoint yet.

Reply:
I think then hide any of the design ids, and cross the bridge when we have a use case for it.  For example, for v2 we're removing the partner data injected keys and maybe NB was relying on those ones however let's see what they need cause it might be just they can actually use the info on the job itself. 
In summary, if it doesn't make sense and it can be confusing let's hide it!